esp/esp/customforms/views.py

```diff
@@ -84,55 +85,73 @@ def onSubmit(request):
-            for section in page['sections']:
-                new_section = Section.objects.create(page=new_page, title=section['data']['question_text'],
-                    description=section['data']['help_text'], seq=int(section['data']['seq'])
+            try:
+                metadata = json.loads(request.body)
+                fields = []
+
+                # truncating field lengths to the character limits specified
+                title = metadata['title'][0:Form._meta.get_field('title').max_length]
+                link_type = metadata['link_type'][0:Form._meta.get_field('link_type').max_length]
+                perms = metadata['perms'][0:Form._meta.get_field('perms').max_length]
+                success_message = metadata['success_message'][0:Form._meta.get_field('success_message').max_length]
+                success_url = metadata['success_url'][0:Form._meta.get_field('success_url').max_length]
+
+                # Creating form
+                form = Form.objects.create(title=title,
+                    description=metadata['desc'], created_by=request.user, link_type=link_type,
+                    link_id=int(metadata['link_id']), anonymous=metadata['anonymous'], perms=perms,
+                    success_message=success_message, success_url=success_url
                     )
 
-                # inserting fields
-                for field in section['fields']:
-                    if 'required' in field['data'] and field['data']['required'] == 'checked':
-                        is_required = True
+                # Set up tag to associate form with registration module
+                if 'link_module' in metadata:
+                    try:
+                        prog = Program.objects.get(id=metadata['link_id'])
+                    except Program.DoesNotExist:
+                        return ESPError('No program with ID %i' % (metadata['link_id']))
+                    if not prog.hasModule(metadata['link_module']):
+                        return ESPError('Program does not have %s enabled' % (metadata['link_module']))
+                    if metadata['link_module'] == 'CustomFormModule':
+                        Tag.setTag(key=metadata['link_tl'] + '_extraform_id', value=form.id, target=prog)
+                    elif metadata['link_module'] == 'TeacherQuizModule':
+                        Tag.setTag(key='quiz_form_id', value=form.id, target=prog)
                     else:
-                        is_required = False
+                        return ESPError('Module %s does not use a custom form or is not implemented' % (metadata['link_module']))
 
-                    new_field = Field.objects.create(form=form, section=new_section, field_type=field['data']['field_type'],
-                        seq=int(field['data']['seq']), label=field['data']['question_text'], help_text=field['data']['help_text'],
-                        required=is_required
-                        )
+                # Inserting pages
```

Comment:
In a similar vein, these three fields should really not have the little checkbox for "required" since they're not true fields and i presume required vs not required would have no effect here.